amqp_socket.c error on windows

![Image](https://github.com/user-attachments/assets/006f9dbb-44d9-426f-a6a5-6326c16f15de)
need fix
![Image](https://github.com/user-attachments/assets/77abe478-d58f-46d1-b047-e57372604e6f)

<details><summary>ioctlsocket</summary>

```
#22 1.201 [INFO]  ### 🚀 STARTING STAGE: 33-librabbitmq-test
#22 1.203 [INFO]  ### DATE: Mon Apr 20 22:20:56 UTC 2026
#22 1.203 [INFO]  ### Starting build function: ffbuild_dockerbuild
#22 1.203 [DEBUG] Verbose mode active. Build output will be shown in real-time.
#22 1.444 -- The C compiler identification is GNU 15.2.0
#22 1.464 -- Detecting C compiler ABI info
#22 1.515 -- Detecting C compiler ABI info - done
#22 1.537 -- Check for working C compiler: /usr/local/bin/x86_64-w64-mingw32-gcc - skipped
#22 1.537 -- Detecting C compile features
#22 1.538 -- Detecting C compile features - done
#22 1.550 -- Looking for poll
#22 1.597 -- Looking for poll - not found
#22 1.598 -- Looking for clock_gettime in rt
#22 1.677 -- Looking for clock_gettime in rt - not found
#22 1.677 -- Looking for posix_spawnp in rt
#22 1.748 -- Looking for posix_spawnp in rt - not found
#22 1.758 -- Found OpenSSL: /opt/ffbuild/lib/libcrypto.a (found version "4.1.0")
#22 1.760 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
#22 1.857 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
#22 1.859 -- Found Threads: TRUE
#22 1.859 -- Looking for ENGINE_new
#22 2.047 -- Looking for ENGINE_new - not found
#22 2.054 -- Performing Test COMPILER_HAS_DEPRECATED_ATTR
#22 2.130 -- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
#22 2.134 -- Building rabbitmq as a shared library - no
#22 2.134 -- Building rabbitmq as a static library - yes
#22 2.136 -- Configuring done (0.8s)
#22 2.154 -- Generating done (0.0s)
#22 2.155 -- Build files have been written to: /build/33-librabbitmq-test/build
#22 3.474 [1/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_api.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_api.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_api.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_api.c
#22 3.541 [2/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_consumer.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_consumer.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_consumer.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_consumer.c
#22 3.606 [3/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_mem.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_mem.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_mem.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_mem.c
#22 3.651 [4/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_connection.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_connection.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_connection.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_connection.c
#22 3.896 [5/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_openssl.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_openssl.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_openssl.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_openssl.c
#22 4.839 [6/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_openssl_bio.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_openssl_bio.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_openssl_bio.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_openssl_bio.c
#22 4.850 [7/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_socket.c
#22 4.850 FAILED: [code=1] librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj 
#22 4.850 /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_socket.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_socket.c
#22 4.850 /build/33-librabbitmq-test/librabbitmq/amqp_socket.c: In function 'connect_socket':
#22 4.850 /build/33-librabbitmq-test/librabbitmq/amqp_socket.c:308:52: error: passing argument 3 of 'ioctlsocket' from incompatible pointer type [-Wincompatible-pointer-types]
#22 4.850   308 |   if (SOCKET_ERROR == ioctlsocket(sockfd, FIONBIO, &one)) {
#22 4.850       |                                                    ^~~~
#22 4.850       |                                                    |
#22 4.850       |                                                    int *
#22 4.850 In file included from /build/33-librabbitmq-test/librabbitmq/amqp_private.h:30,
#22 4.850                  from /build/33-librabbitmq-test/librabbitmq/amqp_socket.c:12:
#22 4.850 /opt/ct-ng/x86_64-w64-mingw32/sysroot/mingw/include/winsock2.h:1007:76: note: expected 'u_long *' {aka 'long unsigned int *'} but argument is of type 'int *'
#22 4.850  1007 |   WINSOCK_API_LINKAGE int WSAAPI ioctlsocket(SOCKET s,__LONG32 cmd,u_long *argp);
#22 4.850       |                                                                    ~~~~~~~~^~~~
#22 5.062 [8/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_framing.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_framing.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_framing.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_framing.c
#22 5.093 [9/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_tcp_socket.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_tcp_socket.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_tcp_socket.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_tcp_socket.c
#22 5.106 [10/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_time.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_time.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_time.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_time.c
#22 5.223 [11/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_table.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_table.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_table.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_table.c
#22 5.501 [12/25] /usr/local/bin/x86_64-w64-mingw32-gcc --sysroot=/opt/ct-ng/x86_64-w64-mingw32/sysroot -DAMQP_STATIC -DHAVE_CONFIG_H -I/build/33-librabbitmq-test/build/librabbitmq/../include -I/build/33-librabbitmq-test/librabbitmq/../include -I/build/33-librabbitmq-test/build/librabbitmq -I/build/33-librabbitmq-test/librabbitmq -I/build/33-librabbitmq-test/librabbitmq/unix -march=broadwell -mtune=broadwell -mavx2 -mfma -O3 -pipe -mms-bitfields -fstack-protector-strong -fopenmp -ffunction-sections -fdata-sections -std=gnu11 -I/opt/ffbuild/include -D__USE_MINGW_ANSI_STDIO=1 -U_WIN32_WINNT -D_WIN32_WINNT=0x0A00 -D_WIN32 -D_FORTIFY_SOURCE=2 -O3 -DNDEBUG -std=gnu99 -fvisibility=hidden -MD -MT librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_url.c.obj -MF librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_url.c.obj.d -o librabbitmq/CMakeFiles/rabbitmq-static.dir/amqp_url.c.obj -c /build/33-librabbitmq-test/librabbitmq/amqp_url.c
#22 5.501 ninja: build stopped: subcommand failed.
#22 5.503 [ERROR] ❌ Build FAILED for 33-librabbitmq-test after 00h:00m:06s
```
</details>

--- a/librabbitmq/amqp_socket.c
+++ b/librabbitmq/amqp_socket.c
@@ -291,6 +291,7 @@ int amqp_open_socket_noblock(char const *hostname, int portnumber,
 #ifdef _WIN32
 static int connect_socket(struct addrinfo *addr, amqp_time_t deadline) {
   int one = 1;
+  u_long nonblocking = 1;
   SOCKET sockfd;
   int last_error;
 
@@ -305,7 +306,7 @@ static int connect_socket(struct addrinfo *addr, amqp_time_t deadline) {
   }
 
   /* Set the socket to be non-blocking */
-  if (SOCKET_ERROR == ioctlsocket(sockfd, FIONBIO, &one)) {
+  if (SOCKET_ERROR == ioctlsocket(sockfd, FIONBIO, &nonblocking)) {
     last_error = AMQP_STATUS_SOCKET_ERROR;
     goto err;
   }

